compiler/src/compiler/string_to_rcst.rs

```diff
@@ -601,18 +624,99 @@ mod parse {
         );
     }
 
+    #[instrument(level = "trace")]
+    fn text_placeholder(input: &str, curly_brace_count: usize) -> Option<(&str, Rcst)> {
+        let (mut input, opening_curly_braces) =
+            parse_multiple(input, opening_curly_brace, Some(curly_brace_count))?;
+
+        let mut expression_text = vec![];
+        let closing_curly_braces = loop {
+            match input.chars().next() {
+                Some('}') => {
+                    match parse_multiple(input, closing_curly_brace, Some(curly_brace_count)) {
+                        Some((input_after_closing_curly_braces, closing_curly_braces)) => {
+                            input = input_after_closing_curly_braces;
+                            break closing_curly_braces;
+                        }
+                        _ => {
+                            input = &input[1..];
+                            expression_text.push('}');
+                        }
+                    }
+                }
+                Some('\n') | None => {
+                    break vec![Rcst::Error {
+                        unparsable_input: "".to_string(),
+                        error: RcstError::TextPlaceholderNotClosed,
+                    }];
+                }
+                Some(c) => {
+                    input = &input[c.len_utf8()..];
+                    expression_text.push(c);
+                }
+            }
+        };
+
+        // FIXME: Better report the error that the placeholder contains more then just one expression
+        if let Some((expression_text_after_expression, expression)) =
+            expression(&expression_text.iter().join(""), 0, true, true) && expression_text_after_expression.is_empty()
```

Comment:
The arguments for expression might not be chosen well.

Reply:
Resolved by [e32802e](https://github.com/candy-lang/candy/pull/258/commits/e32802e84d9d4e77149b1d7e418235be674f909f)